### PR TITLE
Remove the use of tito

### PR DESCRIPTION
**Pull request by ehelms**
With the change to using Koji CLI directly for build and release,
the tito configuration is unnecessary to continue managing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,9 +23,7 @@ If you're submitting a patch which adds/updates npm modules (nodejs- packages) y
 To build locally or release RPMs from this repo, you also require:
 
 * [obal](https://github.com/theforeman/obal) 0.0.2 or higher
```
Comment (evgeni) [inline README.md:25]: You need an obal release with the un-tito-fication in, which doesn't exist yet ;)

File: package_manifest.yaml [hunk 2]
```diff
@@ -248,14 +240,14 @@ foreman_core_packages:
...
+        - "{{ all_tags['foreman-el8'] }}"
+        - "{{ all_tags['foreman-plugins-el8'] }}"
     rubygem-net_http_unix: {}
     rubygem-netrc:
-      releasers:
-        - koji-foreman
-        - koji-foreman-plugins
+      koji_tags:
+        - "{{ all_tags['foreman-el8'] }}"
+        - "{{ all_tags['foreman-plugins-el8'] }}"
```
Comment (ekohl) [inline package_manifest.yaml:250]: Why do we release these to both foreman and foreman-plugins?

File: list_koji_tags.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env python3
+
+from ruamel.yaml import YAML
```
Comment (ekohl) [inline list_koji_tags.py:3]: If you're not writing YAML, I'd use vanilla yaml
```suggestion
import yaml
```

File: list_koji_tags.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env python3
```
Comment (ekohl) [inline list_koji_tags.py:1]: I'd not use a file extension so you're free to change the implementation language.

File: add_pypi_package.sh [hunk 5]
```diff
@@ -105,13 +94,13 @@ add_pypi_to_comps() {
...
 		local section="foreman_nonscl_packages"
-	elif [[ $TITO_TAG == "foreman-plugins-nightly-nonscl-rhel7" ]] ; then
+	elif [[ $KOJI_TAG == "foreman-plugins-nightly-nonscl-rhel7" ]] ; then
 		local section="plugin_nonscl_packages"
-	elif [[ $TITO_TAG == "katello-nightly-rhel7" ]] ; then
+	elif [[ $KOJI_TAG == "katello-nightly-rhel7" ]] ; then
 		local section="katello_packages"
-	elif [[ $TITO_TAG == "katello-pulpcore-nightly-el7" ]] ; then
+	elif [[ $KOJI_TAG == "katello-pulpcore-nightly-el7" ]] ; then
 		local section="pulpcore_packages"
```
Comment (ekohl) [inline add_pypi_package.sh:104]: This should be updated to EL8. Alternatively, we can not talk about a koji tag but require the user to specify the section in the manifest.

**Review by ekohl — COMMENTED** [review]
So how do we deal with package whitelisting in Koji after this?

**Comment by ehelms** [comment]
> So how do we deal with package whitelisting in Koji after this?

Checking the status of? Actually doing it? What do you mean specifically here?

File: add_pypi_package.sh (hunk 5 above)
Comment (ehelms) [reply add_pypi_package.sh:104]: I thought about this as well and figured it as a future update to persue on top of this change.

**Comment by ehelms** [comment]
> Do we continue using the whitelist? If so, how do we ensure new packages are whitelisted?
> 
> I enjoyed using `rel-eng/compare-to-tags` to see if the whitelists were correct. If we continue using whitelists, I'd like to see a replacement.

I am not sure *which* whitelist you mean here: koji or tito. Koji enforces whitelisting as packages must be added to a tag. We are not managing that in anyway today to my knowledge. Tito whitelisting was a filtering mechanism for identifying which packages should be built against what tags -- this is handled by `koji_tags` property with Obal.

File: package_manifest.yaml (hunk 2 above)
Comment (ehelms) [reply package_manifest.yaml:250]: Something something SCl...goes looking.... "Also build rubygem-net-ssh for non-SCL " so I think this means we can stop this.

**Comment by ekohl** [comment]
Right, but the script `rel-eng/compare-to-tags` gave an overview of which Koji packages were whitelisted but shouldn't according to the tito config and vice versa: which weren't whitelisted but should. This helped to clean things up.

So what I'm essentially asking for (since we'll keep the koji whitelist) is a way to gather what should be in each koji tag and then diff it against what's actually in the tag. Does that make sense?

**Comment by ehelms** [comment]
> Right, but the script `rel-eng/compare-to-tags` gave an overview of which Koji packages were whitelisted but shouldn't according to the tito config and vice versa: which weren't whitelisted but should. This helped to clean things up.
> 
> So what I'm essentially asking for (since we'll keep the koji whitelist) is a way to gather what should be in each koji tag and then diff it against what's actually in the tag. Does that make sense?

Last time this was requested the answer I put together was -- https://github.com/theforeman/obal/pull/243

The other idea I had been thinking about was moving tag management into this repository from tool_belt and using that to declare and manage the Koji whitelist tightly -- https://github.com/ktdreyer/koji-ansible#koji_tag

**Comment by ehelms** [comment]
> Last time this was requested the answer I put together was -- [theforeman/obal#243](https://github.com/theforeman/obal/pull/243)

Using this functionality locally, as the solution you'd get the following output right now (which identifies a few actions):

```
obal-source verify-koji-tag all

PLAY [localhost] *************************************************************************************************************************************************************************************************************************************************************************
included: /home/ehelms/workspace/upstream/packaging/obal/obal/data/roles/verify_koji_tag/tasks/all.yaml for localhost

TASK [verify_koji_tag : Find all tags] ***************************************************************************************************************************************************************************************************************************************************
ok: [localhost]
included: /home/ehelms/workspace/upstream/packaging/obal/obal/data/roles/verify_koji_tag/tasks/verify.yaml for localhost
included: /home/ehelms/workspace/upstream/packaging/obal/obal/data/roles/verify_koji_tag/tasks/verify.yaml for localhost
included: /home/ehelms/workspace/upstream/packaging/obal/obal/data/roles/verify_koji_tag/tasks/verify.yaml for localhost
included: /home/ehelms/workspace/upstream/packaging/obal/obal/data/roles/verify_koji_tag/tasks/verify.yaml for localhost
included: /home/ehelms/workspace/upstream/packaging/obal/obal/data/roles/verify_koji_tag/tasks/verify.yaml for localhost
included: /home/ehelms/workspace/upstream/packaging/obal/obal/data/roles/verify_koji_tag/tasks/verify.yaml for localhost

TASK [verify_koji_tag : Verify tag] ******************************************************************************************************************************************************************************************************************************************************
changed: [localhost]

TASK [verify_koji_tag : Packages in Koji but not in git for foreman-client-nightly-el8] **************************************************************************************************************************************************************************************************
ok: [localhost] => 
  msg:
  - jsoncpp
  - qpid-proton

TASK [verify_koji_tag : Verify tag] ******************************************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [verify_koji_tag : Print packages missing in koji for foreman-client-nightly-el9] ***************************************************************************************************************************************************************************************************
ok: [localhost] => 
  msg: Tag foreman-client-nightly-el9 verified

TASK [verify_koji_tag : Verify tag] ******************************************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [verify_koji_tag : Print packages missing in koji for foreman-client-nightly-rhel7] *************************************************************************************************************************************************************************************************
ok: [localhost] => 
  msg: Tag foreman-client-nightly-rhel7 verified

TASK [verify_koji_tag : Verify tag] ******************************************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [verify_koji_tag : Print packages missing in koji for foreman-nightly-el8] **********************************************************************************************************************************************************************************************************
ok: [localhost] => 
  msg: Tag foreman-nightly-el8 verified

TASK [verify_koji_tag : Verify tag] ******************************************************************************************************************************************************************************************************************************************************
changed: [localhost]

TASK [verify_koji_tag : Packages in Koji but not in git for foreman-plugins-nightly-el8] *************************************************************************************************************************************************************************************************
ok: [localhost] => 
  msg:
  - yggdrasil

TASK [verify_koji_tag : Verify tag] ******************************************************************************************************************************************************************************************************************************************************
changed: [localhost]

TASK [verify_koji_tag : Packages in Koji but not in git for katello-nightly-el8] *********************************************************************************************************************************************************************************************************
ok: [localhost] => 
  msg:
  - rubygem-pulp_2to3_migration_client
  - rubygem-redhat_access
  - rubygem-redhat_access_lib

TASK [verify_koji_tag : Packages missing in koji for katello-nightly-el8] ****************************************************************************************************************************************************************************************************************
ok: [localhost] => 
  msg:
  - rubygem-qpid_proton
```

**Comment by ekohl** [comment]
I know you wanted to get this in before 3.4 branched, but I haven't gotten around to it. Can we get RC1 out and then see how this goes in nightly? I'd be happy to cherry pick it before RC2 if it works well.

**Comment by ehelms** [comment]
> I know you wanted to get this in before 3.4 branched, but I haven't gotten around to it. Can we get RC1 out and then see how this goes in nightly? I'd be happy to cherry pick it before RC2 if it works well.

Sure? Everything in here related to tito is just noise in the repository at this point as tito is not used in any of the build paths. The other changes make things cleaner / nicer or more accurate with the current source of truth.

**Comment by ehelms** [comment]
The corresponding Obal functionality for tag verification has been merged.

**Comment by ehelms** [comment]
Rebased and cleaned this up

**Review by ekohl — CHANGES_REQUESTED** [review]
I have since introduced https://github.com/theforeman/foreman-packaging/blob/rpm/develop/remove_package. Could you also modify that?

File: README.md [hunk 6]
```diff
@@ -46,27 +44,17 @@ Configuration for mock is supplied in mock/ and can be used to build any of
...
+then you can use Obal to build scratch packages on Koji, though it's slower
 than mock (above).
 
     obal scratch PACKAGE
 
-Using a local git checkout, change `source_dir` as appropriate:
-
-* Core packages: `obal scratch foreman -e "{'releasers':['koji-foreman-jenkins'], 'build_package_tito_releaser_args': ['--arg source_dir=~/foreman']}"`
-* Plugins: `obal scratch rubygem-foreman_bootdisk -e "{'releasers': ['koji-foreman-plugins-jenkins']}, 'build_package_tito_releaser_args': ['--arg source_dir=~/foreman_bootdisk']}"`
-* Katello: `obal scratch rubygem-katello -e "{'releasers': ['koji-katello-jenkins'], 'build_package_tito_releaser_args': ['--arg source_dir=~/katello']}"`
```
[the remark this answers is not in the document]
Comment (ehelms) [reply README.md:68]: That is correct, that is functionality that would need to be added to Obal.

File: add_gem_package.sh [hunk 7]
```diff
@@ -1,9 +1,9 @@
 #!/bin/bash -e
 
 usage() {
-	echo "Usage: $0 GEM_NAME [TEMPLATE [TITO_TAG [PACKAGE_SUBDIR]]]"
+	echo "Usage: $0 GEM_NAME [TEMPLATE [KOJI_TAG [PACKAGE_SUBDIR]]]"
```
[the remark this answers is not in the document]
Comment (ehelms) [reply add_gem_package.sh]: Given the `package_manifest` sections can be changed or split out into smaller ones that make sense to update or group together and I am hesitant to tie the updates to that just yet. I'd rather build up more first class support for identifying and calculating it.

**Comment by ehelms** [comment]
Rebased on latest breaking changes, updates to `rel-eng` will keep breaking this PR so I'd like to merge it.

**Comment by ehelms** [comment]
Rebased again - any final comments before I merge?

**Pull request merged**